apis/duck/v1/destination_test.go

```diff
@@ -183,3 +197,57 @@ func TestDestination_GetRef(t *testing.T) {
 		})
 	}
 }
+
+func TestDestinationSetDefaults(t *testing.T) {
+	ctx := context.TODO()
+
+	parentNamespace := "parentNamespace"
+
+	tests := map[string]struct {
+		d    *Destination
+		ctx  context.Context
+		want string
+	}{
+		"uri set, nothing in ref, not modified ": {
+			d:    &Destination{URI: apis.HTTP("example.com")},
+			ctx:  ctx,
+			want: "",
+		},
+		"namespace set, nothing in context, not modified ": {
+			d:    &Destination{Ref: &KnativeReference{Namespace: namespace}},
+			ctx:  ctx,
+			want: namespace,
+		},
+		"namespace set, context set, not modified ": {
+			d:    &Destination{Ref: &KnativeReference{Namespace: namespace}},
+			ctx:  apis.WithinParent(ctx, metav1.ObjectMeta{Namespace: parentNamespace}),
+			want: namespace,
+		},
+		"namespace set, uri set, context set, not modified ": {
+			d:    &Destination{Ref: &KnativeReference{Namespace: namespace}, URI: apis.HTTP("example.com")},
+			ctx:  apis.WithinParent(ctx, metav1.ObjectMeta{Namespace: parentNamespace}),
+			want: namespace,
+		},
+		"namespace not set, context set, defaulted": {
+			d:    &Destination{Ref: &KnativeReference{}},
+			ctx:  apis.WithinParent(ctx, metav1.ObjectMeta{Namespace: parentNamespace}),
+			want: parentNamespace,
+		},
+		"namespace not set, uri set, context set, defaulted": {
+			d:    &Destination{Ref: &KnativeReference{}, URI: apis.HTTP("example.com")},
+			ctx:  apis.WithinParent(ctx, metav1.ObjectMeta{Namespace: parentNamespace}),
+			want: parentNamespace,
+		},
+	}
+	for name, tc := range tests {
+		t.Run(name, func(t *testing.T) {
+			tc.d.SetDefaults(tc.ctx)
+			if tc.d.Ref != nil && tc.d.Ref.Namespace != tc.want {
+				t.Errorf("%s: got: %s wanted %s", name, tc.d.Ref.Namespace, tc.want)
```

Comment:
```suggestion
				t.Errorf("Got: %s wanted %s", tc.d.Ref.Namespace, tc.want)
```

Reply:
it is helpful to have the test name in the test output

Reply:
When you use `t.Run` the test name is right there, above :-)